beamer/tests/agent/multiple_processors/test_slave.py

```diff
@@ -0,0 +1,32 @@
+import json
+import os
+import sys
+import time
+import ape
+
+from beamer.tests.util import alloc_accounts, make_request
+
+
+def test_sending_request_slave(contracts, token, finality_period):
+    requester, target = alloc_accounts(2)
+    chain_id = ape.chain.chain_id
+    request_count = int(os.environ.get("REQUEST_COUNT", 0))
+    for i in range(1, request_count + 1):
+        if i == chain_id:
+            continue
+        contracts.request_manager.setFinalityPeriod(i, finality_period)
+        make_request(contracts.request_manager, token, requester, target, 1, target_chain_id=i)
+    print("Chain is ready", flush=True)
+
+    while True:
+        command = sys.stdin.readline()
+        if command == "get_contracts\n":
+            contract_addresses = {
+                "request_manager": contracts.request_manager.address,
+                "fill_manager": contracts.fill_manager.address,
+                "token": token.address,
+                "l2_messenger": contracts.l2_messenger.address,
+            }
+            sys.stdout.write(json.dumps(contract_addresses) + "\n")
+            sys.stdout.flush()
```

Comment:
Why write `\n` if you're going to call `flush`?

Reply:
first iteration was like that and without \n flush didnt work.

Reply:
> first iteration was like that and without \n flush didnt work.

What is the buffering mode of `sys.stdout` here?